ui/app/pages/asset/components/native-asset.js

```diff
@@ -1,19 +1,30 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import { useHistory } from 'react-router-dom';
-
 import TransactionList from '../../../components/app/transaction-list';
 import { EthOverview } from '../../../components/app/wallet-overview';
-import { getSelectedIdentity } from '../../../selectors/selectors';
+import {
+  getSelectedIdentity,
+  getCurrentChainId,
+  getRpcPrefsForCurrentProvider,
+} from '../../../selectors/selectors';
+import { showModal } from '../../../store/actions';
+import getAccountLink from '../../../../lib/account-link';
 import { DEFAULT_ROUTE } from '../../../helpers/constants/routes';
-
 import AssetNavigation from './asset-navigation';
+import TokenOptions from './token-options';
 
 export default function NativeAsset({ nativeCurrency }) {
   const selectedAccountName = useSelector(
     (state) => getSelectedIdentity(state).name,
   );
+  const dispatch = useDispatch();
+
+  const chainId = useSelector(getCurrentChainId);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider);
+  const selectedIdentity = useSelector(getSelectedIdentity);
```

Comment:
There is also `getSelectedAddress` selector which just returns the address. The getSelectedIdentity selector first calls getSelectedAddress then gets the full identity object by address. Might be worthwhile to use `getSelectedAddress` here if we don't expect to use any of the identity fields.

Reply:
Thanks! Will do!